src/test/txreconciliation_tests.cpp

```diff
@@ -81,4 +81,60 @@ BOOST_AUTO_TEST_CASE(IsPeerRegisteredTest)
     BOOST_CHECK(!tracker.IsPeerRegistered(peer_id0));
 }
 
+BOOST_AUTO_TEST_CASE(AddToSetTest)
+{
+    TxReconciliationTracker tracker(TXRECONCILIATION_VERSION);
+    NodeId peer_id0 = 0;
+    FastRandomContext frc{/*fDeterministic=*/true};
+
+    Wtxid wtxid{Wtxid::FromUint256(frc.rand256())};
+
+    BOOST_REQUIRE(!tracker.IsPeerRegistered(peer_id0));
+    BOOST_REQUIRE(!tracker.AddToSet(peer_id0, wtxid));
+
+    tracker.PreRegisterPeer(peer_id0);
+    BOOST_REQUIRE_EQUAL(tracker.RegisterPeer(peer_id0, true, 1, 1), ReconciliationRegisterResult::SUCCESS);
+    BOOST_CHECK(tracker.IsPeerRegistered(peer_id0));
+
+    BOOST_REQUIRE(tracker.AddToSet(peer_id0, wtxid));
+
+    tracker.ForgetPeer(peer_id0);
+    Wtxid wtxid2{Wtxid::FromUint256(frc.rand256())};
+    BOOST_REQUIRE(!tracker.AddToSet(peer_id0, wtxid2));
+
+    NodeId peer_id1 = 1;
+    tracker.PreRegisterPeer(peer_id1);
+    BOOST_REQUIRE_EQUAL(tracker.RegisterPeer(peer_id1, true, 1, 1), ReconciliationRegisterResult::SUCCESS);
+    BOOST_CHECK(tracker.IsPeerRegistered(peer_id1));
+
+    for (size_t i = 0; i < 3000; ++i)
```

Comment:
In "p2p: Functions to add/remove wtxids to tx reconciliation sets" 2e29e9beeebcadfc7afc56a28791456d40551bed: I think `MAX_SET_SIZE` could be in `txreconciliation.h` then we can use it in the tests.

```diff
-/**
- * Maximum number of wtxids stored in a peer local set, bounded to protect the memory use of
- * reconciliation sets and short ids mappings, and CPU used for sketch computation.
- */
-constexpr size_t MAX_SET_SIZE = 3000;
 
 /**
  * Maximum number of transactions for which we store assigned fanout targets.
diff --git a/src/node/txreconciliation.h b/src/node/txreconciliation.h
index c2cdf9875f..648b6369c0 100644
--- a/src/node/txreconciliation.h
+++ b/src/node/txreconciliation.h
@@ -14,6 +14,12 @@
 /** Supported transaction reconciliation protocol version */
 static constexpr uint32_t TXRECONCILIATION_VERSION{1};
 
+/**
+ * Maximum number of wtxids stored in a peer local set, bounded to protect the memory use of
+ * reconciliation sets and short ids mappings, and CPU used for sketch computation.
+ */
+constexpr size_t MAX_SET_SIZE = 3000;
+
 enum class ReconciliationRegisterResult {
     NOT_FOUND,
     SUCCESS,
diff --git a/src/test/txreconciliation_tests.cpp b/src/test/txreconciliation_tests.cpp
index a0071e4388..8befe0d36e 100644
--- a/src/test/txreconciliation_tests.cpp
+++ b/src/test/txreconciliation_tests.cpp
@@ -115,7 +115,7 @@ BOOST_AUTO_TEST_CASE(AddToSetTest)
     BOOST_REQUIRE_EQUAL(tracker.RegisterPeer(peer_id1, true, 1, 1), ReconciliationRegisterResult::SUCCESS);
     BOOST_CHECK(tracker.IsPeerRegistered(peer_id1));
 
-    for (size_t i = 0; i < 3000; ++i)
+    for (size_t i = 0; i < MAX_SET_SIZE; ++i)
         BOOST_REQUIRE(tracker.AddToSet(peer_id1, Wtxid::FromUint256(frc.rand256())));
     BOOST_REQUIRE(!tracker.AddToSet(peer_id1, Wtxid::FromUint256(frc.rand256())));
```

Reply:
I've been talking to @sipa about this recently and I think the limit may not be necessary if the approach for when to compute the reconciliation sets is changed. I'm may get rid of it, but if not, I'll take this

Reply:
Ended up taking this, since the limit will be kept in the end.

Thanks!